src/zenml/artifacts/external_artifact_config.py

```diff
@@ -42,12 +42,33 @@ class ExternalArtifactConfiguration(BaseModel):
     pipeline_name: Optional[str] = None
     artifact_name: Optional[str] = None
     model_name: Optional[str] = None
-    model_version: Optional[str] = None
-    model_artifact_name: Optional[Union[str, ModelStages]] = None
+    model_version_name: Optional[Union[str, ModelStages]] = None
+    model_version_number: Optional[int] = None
+    model_artifact_name: Optional[str] = None
     model_artifact_version: Optional[str] = None
     model_artifact_pipeline_name: Optional[str] = None
     model_artifact_step_name: Optional[str] = None
 
+    @root_validator
+    def _validate_all(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        """Validates all fields.
+
+        Args:
+            values: The values dict used to instantiate the model.
+
+        Returns:
+            The validated values dict.
+        """
+        if values.get("model_version_number", None) and values.get(
+            "model_version_name", None
+        ):
+            logger.warning(
+                "`model_version_number` has higher priority then `model_version_name`."
+                "Setting `model_version_name` to `None`."
+            )
+            values["model_version_name"] = None
+        return values
```

Comment:
IMO this feels a bit clunky - why does the user need to specify whether it's a number or name in the first place?

Reply:
I simplified it 👍🏼